CUJExecutor/app/src/androidTest/java/com/example/firstapp/ShellUtility.java

```diff
@@ -144,6 +148,22 @@ void executeUncachedAction() throws UiObjectNotFoundException, invalidInputExcep
             } else {
                 object2 = device.wait(Until.findObject(By.descContains(description)), actionTimeout);
             }
+            if (object2 == null) {
+                List<UiObject2> views = device.wait(Until.findObjects(By.pkg(curPackage)), actionTimeout);
+                if (views == null) {
+                    throw new invalidInputException("no image matching description \"" + description + "\' found," +
+                            " and no Images were were found either... Try something else.");
+                }
+                List<String> possibleDescriptions = new ArrayList<>();
+                for (int i = 0; i < views.size(); i++) {
+                    UiObject2 object = views.get(i);
+                    if (object.getText() == null && object.getContentDescription() != null) {
+                        possibleDescriptions.add(object.getContentDescription());
+                    }
+                }
+                throw new invalidInputException("no image matching description \"" + description + "\' found," +
+                        " but images with the following descriptions were found:\n" + possibleDescriptions + "\nTry again with one of these.");
+            }
```

Comment:
1. `invalidInputException` should be capitalized

2. (optional) may want to move this to a helper method to make it easier to read the code for happy-path executions